src/lib/instrument-listen.js

```diff
@@ -11,8 +14,16 @@ const path = require('path');
 const fs = require('fs');
 const readFile = denodeify(fs.readFile);
 
-module.exports = (app, meta, initPromises) => {
-	async function createServer(app) {
+module.exports = class InstrumentListen {
+	constructor(app, meta, initPromises) {
+		this.app = app;
+		/** @type {TickingMetric[]} */
+		this.tickingMetrics = [];
+		this.server = null;
+		this.initApp(meta, initPromises);
+	}
+
+	async createServer(app) {
```

Comment:
**nitpick:** now that the `app` is stored as a class member in `this.app` we don't need to pass it in to this method

Reply:
good spot! have updated to do this